addons/website_sale/controllers/main.py

```diff
@@ -1249,6 +1249,10 @@ def shop_payment_transaction(self, order_id, access_token, **kwargs):
         except AccessError:
             raise ValidationError("The access token is invalid.")
 
+        order = request.env['sale.order'].browse(order_id)
+        if order.state == 'cancel':
+            raise ValidationError('The order has been canceled.')
```

Comment:
`_document_check_access` already returns the **sudoed** record. You can then simply do:
```suggestion
        if order_sudo.state == 'cancel':
            raise ValidationError('The order has been canceled.')
```
Note that the return value of the method is labelled `order_sudo` as a convention to immediately see which recordset's environment is sudoed or not later in the code.

Reply:
Thanks for the tip ;)